src/sentry/models/groupsearchview.py

```diff
@@ -0,0 +1,43 @@
+from django.db import models
+from django.db.models import UniqueConstraint
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import region_silo_model
+from sentry.db.models.base import DefaultFieldsModel
+from sentry.db.models.fields.foreignkey import FlexibleForeignKey
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+from sentry.models.savedsearch import SortOptions
+
+
+@region_silo_model
+class GroupSearchView(DefaultFieldsModel):
+    """
+    A model for a user's view of the issue stream
+    """
+
+    __relocation_scope__ = RelocationScope.Organization
+    name = models.TextField(max_length=128)
+    user_id = HybridCloudForeignKey("sentry.User", on_delete="CASCADE", null=True)
+    organization = FlexibleForeignKey("sentry.Organization", null=True)
+
+    query = models.TextField()
+    query_sort = models.CharField(
+        max_length=16, default=SortOptions.DATE, choices=SortOptions.as_choices(), null=True
+    )
+    position = models.PositiveSmallIntegerField()
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_groupsearchview"
+        unique_together = ()
+        # Two views cannot occupy the same position in an organization user's list of views
+        constraints = [
+            UniqueConstraint(
+                fields=["user_id", "organization_id", "position"],
+                name="sentry_issueviews_unique_view_position_per_org_user",
+            )
+        ]
```

Comment:
You might need an additional unique here that's basically `organization_id, position where user_id is null`